hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java

```diff
@@ -0,0 +1,505 @@
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount - amountToBurn),
+                        // Create a new key with treshhold key with correct contract id
+                        newKeyNamed(TRESHOLD_KEY_CORRECT_CONTRACT_ID)
+                                .shape(TRESHOLD_KEY_SHAPE.signedWith(sigs(ON, BURN_TOKEN))),
+                        // Update token supply key with the correct contract id
+                        tokenUpdate(FUNGIBLE_TOKEN).supplyKey(TRESHOLD_KEY_CORRECT_CONTRACT_ID),
+                        // Update signer with the correct contract id
+                        cryptoUpdate(SIGNER).key(TRESHOLD_KEY_CORRECT_CONTRACT_ID),
+                        // Test Case 2: the Treasury account is paying and signing a token burn transaction,
+                        // SIGNER → call → CONTRACT → call → PRECOMPILE
+                        contractCall(BURN_TOKEN, "burnToken", BigInteger.valueOf(amountToBurn), new long[0])
+                                .via(SIGNER_BURNS_WITH_SIGNER_PUBLIC_KEY)
+                                .gas(GAS_TO_OFFER)
+                                .payingWith(TOKEN_TREASURY)
+                                .signedBy(TOKEN_TREASURY),
+                        // Assert that the token is burned - total supply should be increased with the amount to burn.
+                        // NOTE: it is multiplied by 2 because of the burned amount in the previous test
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount - 2 * amountToBurn),
+                        // Test Case 3: one account  paying and another one signing a token burn transaction
+                        // SIGNER → call → CONTRACT → call → PRECOMPILE
+                        contractCall(BURN_TOKEN, "burnToken", BigInteger.valueOf(amountToBurn), new long[0])
+                                .via(SIGNER_AND_PAYER_ARE_DIFFERENT)
+                                .gas(GAS_TO_OFFER)
+                                .payingWith(SIGNER2)
+                                .signedBy(SIGNER2, SIGNER),
+                        // Assert that the token is burned - total supply should be increased with the amount to burn.
+                        // NOTE: it is multiplied by 3 because of the burned amount in the previous tests
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount - 3 * amountToBurn),
+                        // Create a key with thresh 1/2 with sigs: new ed25519 key, contractId of burnToken contract
+                        newKeyNamed(TRESHOLD_KEY_CORRECT_CONTRACT_ID)
+                                .shape(TRESHOLD_KEY_SHAPE.signedWith(sigs(ON, BURN_TOKEN))),
+                        // Update the token supply key to with the created key
+                        tokenUpdate(FUNGIBLE_TOKEN).supplyKey(TRESHOLD_KEY_CORRECT_CONTRACT_ID),
+                        // Test Case 4: Signer paying and signing a token burn transaction.
```

Comment:
I think it would be more readable if we leave spaces between the test case comments and the code, because its hard to find where a case ends.

Reply:
Left it like that for now, because if we add empty lines the file will become a lot longer